app/src/App.tsx

```diff
@@ -1,36 +1,63 @@
 import { configureStore } from '@reduxjs/toolkit'
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { Provider } from 'react-redux'
 import { ThemeProvider } from 'styled-components'
 import Web3Provider from 'web3-react'
 
 import 'react-datepicker/dist/react-datepicker.css'
 import 'sanitize.css'
 
+import { MAINNET_LOCATION, XDAI_LOCATION } from './common/constants'
+import { HeaderNoRouter } from './components/common/layout/header'
 import { Main } from './components/main'
+import SettingsViewContainer from './components/settings/settings_view'
 import { ApolloProviderWrapper } from './contexts/Apollo'
 import { ConnectedCPK, ConnectedWeb3 } from './hooks'
 import balanceReducer from './store/reducer'
 import theme from './theme'
 import { GlobalStyle } from './theme/global_style'
 import connectors from './util/connectors'
+import { getInfuraUrl, networkIds } from './util/networks'
+import { checkRpcStatus, getNetworkFromChain } from './util/tools'
 
 const store = configureStore({ reducer: balanceReducer })
 
 const App: React.FC = () => {
+  const windowObj: any = window
+
+  const ethereum = windowObj.ethereum
```

Comment:
This will again run into the same error that we fixed before for non web3 browsers.
![image](https://user-images.githubusercontent.com/4677914/109843687-eb937d80-7c7d-11eb-889a-68fe331c41fa.png)
You should check if etherum object exists before trying to reference ethereum.chainId